README.md

```diff
@@ -1,24 +1,107 @@
+
+### Introduction
+The [Burner Wallet 2.0](https://github.com/burner-wallet/burner-wallet-2) is a modular, extendable and customizable web application.
+It provides a set of packages and plugins that allow you to extend the basic functionality. Here we will focus on the Assets package and the Exchange Plugin.
+
+#### Assets
+You can add any native or ERC20 token to track its balance and send transactions. For example you can define a new ERC20 and add it to the wallet:
+```javascript
+import { xdai, dai, eth, ERC20Asset } from '@burner-wallet/assets';
+
+const test = new ERC20Asset({
+  id: 'tst', // The internal ID used by the wallet.
+  name: 'Test Token', // The display name that will be displayed to the user.
+  network: '1', // The chain ID of the chain the token is deployed to.
+  address: '0x52ad726d80dbb4A9D4430d03657467B99843406b' // Address where the token contract is deployed
+});
+
+const core = new BurnerCore({
+  assets: [test, xdai, dai, eth],
+});
+```
 
-### Clone the sample repo 
-Clone the sample repo to create a plugin project that uses the TokenBridge plugin. This sample repo containes a ready to publish plugin for a ERC677 to ERC677 bridge extension, but you can later modify it to define your own resources.
+The `ERC20Asset` class extends from the `Asset` class. The `Asset` class provides several methods to allow the wallet interact with the token.
+The following are the most important to pay attention and that could be overridden if your token behaves differently.
+- `getBalance`: Indicates how to get the balance of your asset.
+- `startWatchingAddress`: Indicates how to listen to value transfers incoming into your account.
+- `_send`: Indicates how to send value to another account.
+
+#### Exchange Plugin
+The `@burner-wallet/exchange` package is an extendable plugin for implementing asset exchanges and bridges.
+The exchange module accepts an array of trading pairs.
+```
+const exchange = new Exchange({
+  pairs: [new Pair1(), new Pair2()]
+})
+```
 
-#### Project structure
-There are two main folders in the project:
+`Pair` is an abstract class that defines the interface for trading pairs. If you would like to add custom trading functionality, you can extend this class and implement it's methods:
+- `estimateAtoB`: Indicates how to estimate the B Asset the user will receive for the amount of A Asset. For example Fees could be considered here.
+- `estimateBtoA`: Indicates how to estimate the A Asset the user will receive for the amount of B Asset. For example Fees could be considered here.
+- `exchangeAtoB`: Indicates how to send A Asset to be exchanged for B Asset. Here you can add logic to detect exchange finalization for example for a more complex exchange operation like bridges.
+- `exchangeBtoA`: Indicates how to send B Asset to be exchanged for A Asset. Here you can add logic to detect exchange finalization for example for a more complex exchange operation like bridges.
+
+### Project structure
```

Comment:
Consider to move details about the TokenBridge plugin with the link to the repo before the project structure section: which components (`ERC677Asset`, `NativeMediatorAsset`, `Mediator`) it provides to be used by adopters, which interfaces these components provides.